File: app_java/src/main/java/com/ysn/test0410/activity/RecyclerViewActivity.java
package com.ysn.test0410.activity;

import android.os.Bundle;
import android.util.Log;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.google.gson.Gson;
import com.google.gson.JsonObject;
import com.ysn.test0410.R;
import com.ysn.test0410.data.App;
import com.ysn.test0410.data.CustomAdapter;
import com.ysn.test0410.data.Data;

import java.io.BufferedReader;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.Reader;
import java.io.StringWriter;
import java.io.Writer;
import java.nio.charset.StandardCharsets;
import java.util.ArrayList;
import java.util.List;

public class RecyclerViewActivity extends AppCompatActivity {
    private static final String TAG = RecyclerViewActivity.class.getSimpleName();
    private RecyclerView mRecyclerView;
    private CustomAdapter mCustomAdapter;
    private List<App> mAppsList = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_recycler_view);
        init();
    }

    private void init() {
        mRecyclerView = findViewById(R.id.recyclerView);
        // 构造一些数据
        for (int i = 0; i < 50; i++) {
            JsonObject jsonObject = load("contents.json");
            App app = new App();
//            app.appName = "xx应用" + i;
//            app.versionName = "100" + i;
//            app.appSize = 1.5 + i;
//            app.imagePath = "sdcard/AAAPUSH/ic_launcher.jpg";
//            app.appIntroduction = "本应用是" + i;
//            app.timeUpdate = "2023年4月" + i % 30 + "日";
            app.appName = jsonObject.get("appName").toString() + i;
            app.versionName = jsonObject.get("versionName").toString() + i;
            app.appSize = Double.parseDouble(jsonObject.get("appSize").toString()) + i;
            app.imagePath = "sdcard/AAAPUSH/ic_launcher.jpg";
            app.appIntroduction = "本应用是" + i;
            app.timeUpdate = "2023年4月" + i % 30 + "日";
            mAppsList.add(app);
        }
        Data.getInstance().setAppsList(mAppsList);
        mCustomAdapter = new CustomAdapter();
        mRecyclerView.setAdapter(mCustomAdapter);
        GridLayoutManager layoutManager = new GridLayoutManager(RecyclerViewActivity.this, 3); // 每行3个
//        layoutManager.setOrientation(RecyclerView.HORIZONTAL); // 设置横向滚动
        mRecyclerView.setLayoutManager(layoutManager);
    }

    private JsonObject load(String fileName) {
        String json = "";

        try {
//            InputStream inputStream = getResources().openRawResource(id);
            InputStream inputStream = this.getAssets().open(fileName);
            Writer writer = new StringWriter();
            char[] buffer = new char[inputStream.available()];
            Reader reader = new BufferedReader(new InputStreamReader(inputStream, StandardCharsets.UTF_8));
            int n;
            while ((n = reader.read(buffer)) != -1) {
                Log.d(TAG, "load n=" + n);
                writer.write(buffer, 0, n);
            }

            json = writer.toString();
        } catch (Exception e) {
            Log.e(TAG, "load json error: " + e.getMessage());
        }

        return new Gson().fromJson(json, JsonObject.class);
    }
}

File: app_kotlin/src/main/java/com/ysn/app_kotlin/activity/RecyclerViewActivity.kt
package com.ysn.app_kotlin.activity

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.google.gson.Gson
import com.google.gson.JsonObject
import com.ysn.app_kotlin.R
import com.ysn.app_kotlin.data.App
import com.ysn.app_kotlin.data.CustomAdapter
import com.ysn.app_kotlin.data.Data
import java.io.*
import java.nio.charset.StandardCharsets

class RecyclerViewActivity : AppCompatActivity() {
    private val TAG: String = "RecyclerViewActivity"
    private var mRecyclerView: RecyclerView? = null
    private var mCustomAdapter: CustomAdapter? = null
    private val mAppsList: MutableList<App> = ArrayList()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_recycler_view)
        init()
    }

    private fun init() {
        mRecyclerView = findViewById(R.id.recyclerView)
        // 构造一些数据
        for (i in 0..49) {
            val app = App()
            val jsonObject = load("contents.json")
            app.appName = jsonObject!!["appName"].toString() + i
            app.versionName = jsonObject["versionName"].toString() + i
            app.appSize = jsonObject["appSize"].toString().toDouble() + i
            app.imagePath = "sdcard/AAAPUSH/ic_launcher.jpg"
            app.appIntroduction = "本应用是$i"
            app.timeUpdate = "2023年4月" + i % 30 + "日"
            mAppsList.add(app)
        }
        Data.instance.appsList = mAppsList
        mCustomAdapter = CustomAdapter()
        mRecyclerView?.adapter = mCustomAdapter
        val layoutManager = GridLayoutManager(this@RecyclerViewActivity, 3) // 每行3个
        mRecyclerView?.layoutManager = layoutManager
    }

    private fun load(fileName: String): JsonObject? {
        var json = ""
        try {
            val inputStream = this.assets.open(fileName)
            val writer: Writer = StringWriter()
            val buffer = CharArray(inputStream.available())
            val reader: Reader =
                BufferedReader(InputStreamReader(inputStream, StandardCharsets.UTF_8))
            var n: Int
            while (reader.read(buffer).also { n = it } != -1) {
                Log.d(TAG, "load n=$n")
                writer.write(buffer, 0, n)
            }
            json = writer.toString()
        } catch (e: Exception) {
            Log.e(TAG, "load json error: " + e.message)
        }
        return Gson().fromJson(json, JsonObject::class.java)
    }
}